app/src/main/java/com/globant/counter/mvp/presenter/CalculatorPresenter.kt

```diff
@@ -0,0 +1,107 @@
+package com.globant.counter.mvp.presenter
+
+import com.globant.counter.R
+import com.globant.counter.mvp.model.CalculatorModel
+import com.globant.counter.mvp.view.CalculatorView
+import com.globant.counter.rx.events.*
+import com.globant.counter.utils.Constants
+import com.globant.counter.utils.RxBus
+
+class CalculatorPresenter(private val model: CalculatorModel, private val view: CalculatorView) {
+
+    fun init() {
+        val activity = view.activity
+        if (activity != null) {
+
+            // Number pressed event
+            RxBus.subscribe(activity, object : OnNumberButtonPressedBusObserver() {
+                override fun onEvent(value: OnNumberButtonPressed) {
+                    if (model.isResult) {
```

Comment:
so your presenter is more testable please abstract all the logic in ALL the events that you have to methods.